### DISCUSSION TITLE
Preview in editor doesnt display ul/ol

### DISCUSSION THREAD

**Discussion by VrbovskyM** [Alex-D/Trumbowyg#1371] [Q&A] [answered]
I have set up my trumbowyg editor with bunch of plugins. everything seems to work except when I use unordered/ordered list, it doesnt preview in editor, but is actually in the output.
Editor with selected ul
![image](https://user-images.githubusercontent.com/100531156/217811451-ec79af29-93c4-4bbc-8282-8faa48d55c9e.png)
Output:
![image](https://user-images.githubusercontent.com/100531156/217811542-973c7dc2-bdc0-44d5-970f-ddff23d71686.png)

**Comment by Alex-D** [comment]
Hey!
What's the setup? Is the reset CSS option in use there? (Note: I do not recommend it)
Make sure to use identical CSS on the edit and output pages.

**Comment by VrbovskyM** [comment]
Okay so Tailwind is the issue
Add this into input.css, add the class to the div that has trumbowyg editor, and its fixed.
```
@layer base {
  .no-tailwindcss-base ol,
    .no-tailwindcss-base ul {
        list-style: revert;
        margin: revert;
        padding: revert;
    }
  
}
```

**Comment by VrbovskyM** [comment]
Since I have you here. I'd like to ask one more question.
I need to somehow disable the option for images to be pasted into editor. I need to avoid base64 image coding
I removed pasteImage and base64 plugins, but user is still able to paste them in.
I want user to only insert image with these two options
![image](https://user-images.githubusercontent.com/100531156/218132588-283c97a1-601d-462b-98d1-031fe1c8d46d.png)

I read this issue but Im failing to follow your solution. Could you please elaborate ?
Thank you in advance!
https://github.com/Alex-D/Trumbowyg/issues/407

**Comment by Alex-D** [comment, marked as answer]
Maybe try a plugin like that:

```js
(function ($) {
    'use strict';

    $.extend(true, $.trumbowyg, {
        plugins: {
            preventImagePaste: {
                init: function (trumbowyg) {
                    trumbowyg.pasteHandlers.push(function (pasteEvent) {
                        try {
                            var items = (pasteEvent.originalEvent || pasteEvent).clipboardData.items,
                                mustPreventDefault = false,
                                reader;

                            for (var i = items.length - 1; i >= 0; i -= 1) {
                                if (items[i].type.match(/^image\//)) {
                                    mustPreventDefault = true;
                                }
                            }

                            if (mustPreventDefault) {
                                pasteEvent.stopPropagation();
                                pasteEvent.preventDefault();
                            }
                        } catch (c) {
                        }
                    });
                }
            }
        }
    });
})(jQuery);
```

**Reply by VrbovskyM** [reply]
Works great. Thanks!
